sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/DownloadHandler.java

```diff
@@ -205,8 +215,7 @@ private void handleDownloadCompleted(Message message) {
             if (runningBlockDownloads.isEmpty()) {
                 db.downloadDao().updateBlobState(downloadId, BlobTransferState.COMPLETED);
 
-                Message nextMessage = DownloadHandlerMessage.createBlobDownloadCompletedMessage(DownloadHandler.this);
-                nextMessage.sendToTarget();
+                closeContent();
```

Comment:
Depending on when looper really quit - this message either gets ignored because looper quit before processing it OR this message processed again which results in extra DB calls that simply repeat what was already done.